core/base/src/TApplication.cxx

```diff
@@ -468,7 +468,46 @@ void TApplication::GetOptions(Int_t *argc, char **argv)
          } else {
             Warning("GetOptions", "-e must be followed by an expression.");
          }
+      } else if (!strcmp(argv[i], "--")) {
+         TObjString* macro = NULL;
 
+         if (fFiles) {
+            for (int f = 0; f < fFiles->GetEntriesFast(); f++) {
+               TObjString* file = dynamic_cast<TObjString*>(fFiles->At(f));
+
+               if (file->TestBit(kExpression))
+                  continue;
+               if (file->String().EndsWith(".root"))
+                  continue;
+               if (strchr(file->String().Data(), '('))
+                  continue;
+
+               if (macro)
+                  Warning("GetOptions", "-- is used with several macroses. "
+                                        "The arguments will be passed to the last one.");
+
+               macro = file;
+            }
+         }
+
+         if (macro) {
+            argv[i] = null;
```

Comment:
No, it shoud not, because later options participate in `strcmp` ([1]), which doesn't like null pointers. `null` is defined as `""`.

[1] https://github.com/BerserkerTroll/root/blob/385d7c5f7196c9b47f8dbc6d126bc6cd079d7dec/core/base/src/TApplication.cxx#L592